tt_metal/jit_build/data_format.cpp

```diff
@@ -264,22 +261,22 @@ std::vector<DataFormat> get_unpack_dst_formats(
         } else if (int_fpu_en) {
             unpack_dst_format.push_back(src_format);
         } else {
-            if (input_formats[i] == DataFormat::Float32 && !unpack_to_dest_mode.empty() && unpack_to_dest_mode[i] != UnpackToDestMode::Default) {
+            if (input_formats[i] == DataFormat::Float32 && len_unpack_to_dest_mode > i && unpack_to_dest_mode[i] != UnpackToDestMode::Default) {
                 unpack_dst_format.push_back(get_single_unpack_dst_format(input_formats[i], pack_format, DataFormat::Float32));
             } else {
                 unpack_dst_format.push_back(get_single_unpack_dst_format(input_formats[i], pack_format, unpack_cond_dst_format));
             }
         }
     }
     for (int i=0 ; i<NUM_OPERANDS ; i++) {
-        if (param_formats[i] == DataFormat::Float32 && !unpack_to_dest_mode.empty() && unpack_to_dest_mode[NUM_OPERANDS+i] != UnpackToDestMode::Default) {
+        if (param_formats[i] == DataFormat::Float32 && len_unpack_to_dest_mode > (NUM_OPERANDS+i) && unpack_to_dest_mode[NUM_OPERANDS+i] != UnpackToDestMode::Default) {
             unpack_dst_format.push_back(get_single_unpack_dst_format(param_formats[i], pack_format, DataFormat::Float32));
         } else {
             unpack_dst_format.push_back(get_single_unpack_dst_format(param_formats[i], pack_format, unpack_cond_dst_format));
         }
     }
     for (int i=0 ; i<NUM_OPERANDS ; i++) {
-        if (intermed_formats[i] == DataFormat::Float32 && !unpack_to_dest_mode.empty() && unpack_to_dest_mode[3*NUM_OPERANDS+i] != UnpackToDestMode::Default) {
+        if (intermed_formats[i] == DataFormat::Float32 && len_unpack_to_dest_mode > (3*NUM_OPERANDS+i) && unpack_to_dest_mode[3*NUM_OPERANDS+i] != UnpackToDestMode::Default) {
```

Comment:
Maybe put 3*NUM_OPERANDS into another define or a helper function? It sounds like an offset to me